File: dart/lib/src/api/login_api.dart

import 'dart:async';
import 'dart:convert';
import 'dart:html';

import 'package:http/browser_client.dart';
import 'package:http/http.dart';

import 'package:angular_tour_of_heroes/src/api/api.dart';
import 'package:angular_tour_of_heroes/src/model/user.dart';

class LoginApi extends Api{

  static const _loginUrl = Api.prefix + 'api/login'; // URL to web API

  BrowserClient _http;

  LoginApi(this._http);

  Future<User> login(String email, String password) async {
    try {
      final response = await _http.post(_loginUrl+"/",
          headers: headers, body: json.encode({"email": email, "password": password}));
      dynamic data = extractData(response);
      User returnedUser = User.fromJson(data);
      print(returnedUser.toJson());
      return returnedUser;
    } catch (e) {
      throw handleError(e);
    }
  }
}

File: java/src/main/java/api/LoginApi.java
package api;


import com.fasterxml.jackson.core.JsonProcessingException;
import model.User;
import service.UserService;

import javax.inject.Inject;
import javax.ws.rs.*;
import javax.ws.rs.core.Response;
import java.util.Map;
import java.util.logging.Level;
import java.util.logging.Logger;

@Path("login")
public class LoginApi {

    private static final Logger LOGGER = Logger.getLogger(LoginApi.class.getName());
    private UserService userService;

    public LoginApi() { }

    @POST
    @Path("")
    @Produces("application/json")
    public Response login(Map<String, String> credentials){

        String email = credentials.get("email");
        String password = credentials.get("password");

        //User u = userService.login(email, password);


        User u = new User(1,"2","3","4","5","6");

        //ObjectMapper Obj = new ObjectMapper();
        //String json = Obj.writeValueAsString(u);

        LOGGER.log(Level.INFO, "Login");
        return Response.ok().header("Access-Control-Allow-Origin", "*").entity(u.toJson()).build();
    }

    @OPTIONS
    @Path("")
    public Response corsLogin() {
        return Response.ok().header("Access-Control-Allow-Headers", "Content-Type").header("Access-Control-Allow-Origin", "*").build();
    }
}
